### PR TITLE
record number of likes 

### PR DESCRIPTION
**Pull request by beblicarl**
#### Pre-Submission Checklist

-   [x] Your pull request targets the `main` branch.
-   [x] Branch name is descriptive and follows the guidelines [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices).
-   [x] All tests pass . Use `git commit --amend` to amend any fixes.

#### Checklist:

-   [x] All tests are passing.
-   [x] Tested changes locally.
-   [x] Improved documentation or readme where necessary.
-   [x] Your commit messages format follows our commit message guideline [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices#commits).

#### What does this PR do?

https://github.com/tbf-challenge/teamwork-be/issues/307
#### How should this be manually tested?

1. Start the server with `npm run dev`
2. You will see with the logs that the migration has been added to the database

#### What are the relevant issues?

https://github.com/tbf-challenge/teamwork-be/issues/307

### CODE REVIEW

File: migrations/1668360468072_add-trigger-function-to-likesCount-on-posts.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        FROM "post_likes"
+        WHERE "postId" = postLike."postId"
+    )
+    WHERE id = postLike."postId";
+    
+    RETURN postLike;
+    
+    
+    
+    END;`
```
Comment (jocrah) [inline migrations/1668360468072_add-trigger-function-to-likesCount-on-posts.js:34]: lets indent a bit to make the flow a little more readable, like the code in the  if statements and so on
```suggestion
`
   DECLARE postLike post_likes%ROWTYPE;
    BEGIN
    
      IF (TG_OP = 'DELETE') THEN
        postLike = OLD;
      ELSIF (TG_OP = 'INSERT') THEN
        postLike = NEW;
    END IF;

    UPDATE posts
    SET "likesCount" = (
        SELECT COUNT(*) 
        FROM "post_likes"
        WHERE "postId" = postLike."postId"
    )
    WHERE id = postLike."postId";
    
    RETURN postLike;
    
    END;
`
```

Comment (jocrah) [reply migrations/1668360468072_add-trigger-function-to-likesCount-on-posts.js]: i think it had a slight lint issue so kindly pull it locally and fix it

Comment (beblicarl) [reply migrations/1668360468072_add-trigger-function-to-likesCount-on-posts.js]: sure

Comment (jocrah) [reply migrations/1668360468072_add-trigger-function-to-likesCount-on-posts.js]: `1668360468072_add-trigger-function-to-likesCount-on-posts.js` should be `1668360468072_add-trigger-function-to-compute-likesCount-on-posts.js`

Comment (beblicarl) [reply migrations/1668360468072_add-trigger-function-to-likesCount-on-posts.js]: done

Comment (jocrah) [reply migrations/1668360468072_add-trigger-function-to-likesCount-on-posts.js]: 1668360468072_add-trigger-function-to-likesCount-on-posts.js should be 1668360468072_add-trigger-function-to-compute-likesCount-on-posts.js

**Review by jocrah — COMMENTED** [review]
@beblicarl, this just came to mind. we would need to add a test case for this change to ensure it works

perhaps something like ` it('should update likesCount column for post')` where we compare the number of likes in the `post_likes` table to the `likesCount` for that post

**Comment by beblicarl** [comment]
> > yeah that's true.. but we still don't return the likesCount field for the post so I am sure it will be for the new issue you talked about
> 
> okay, however, its not related to whether something is being returned or not. Its mainly checking that an insert/delete updates the `likesCount` in the table. Just like you were manually checking the `posts` table after you inserted/deleted a post_like locally

yes please

**Review by jocrah — APPROVED** [review]

**Pull request merged**